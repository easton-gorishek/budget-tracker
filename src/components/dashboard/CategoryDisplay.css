:local(.categoryDisplay) {
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;

  #categoryHeader {
    color: #14A76C;
    font-size: 1.5em;
    border: thin solid #14A76C;
    border-radius: 5px;
    padding: .25em;
  }
  #categoryEdit {
    align-self: flex-end;
    font-size: 1.25em;
    border-radius: 5px;
  }

  button {
    background: #242323;
    border: none;
    color: #14A76C;
    cursor: pointer;
  }

  button:hover {
    background: #14A76C;
    color: black;
  }
}