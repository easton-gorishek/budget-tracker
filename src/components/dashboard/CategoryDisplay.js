import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './CategoryDisplay.css';
import Expenses from '../expenses/Expenses';

class CategoryDisplay extends Component {
  
  static propTypes = {
    category: PropTypes.object.isRequired,
    onEdit: PropTypes.func.isRequired,
  };

  render() {
    const { category, onEdit } = this.props;

    return (
      <div className={styles.categoryDisplay}>
        <button id="categoryEdit" name="edit" onClick={onEdit}>
          <i className="fas fa-pencil-alt"></i>
        </button>
        <p id="categoryHeader">
          <strong>{category.name}</strong> |&nbsp; 
          <strong>${category.budget}</strong>
        </p>
        <Expenses categoryId={category.id}/>
      </div>

    );
  }
}

export default CategoryDisplay;