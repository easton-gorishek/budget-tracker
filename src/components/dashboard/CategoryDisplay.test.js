import  CategoryDisplay  from './CategoryDisplay';
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

describe('category display', () => {

  it('renders display, then edit', () => {
    const handleEdit = jest.fn();

    const wrapper = shallow(<CategoryDisplay
      category={{ name: 'name', budget: 'budget' }}
      onEdit={handleEdit}
    />);

    expect(toJSON(wrapper)).toMatchSnapshot();

    wrapper.find('button[name="edit"]').simulate('click');
    const editCalls = handleEdit.mock.calls;
    expect(editCalls.length).toBe(1);

  });
});
