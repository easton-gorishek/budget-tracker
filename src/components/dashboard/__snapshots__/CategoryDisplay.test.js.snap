// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`category display renders display, then edit 1`] = `
<div
  className="categoryDisplay"
>
  <button
    id="categoryEdit"
    name="edit"
    onClick={[MockFunction]}
  >
    ✎
  </button>
  <p
    id="categoryHeader"
  >
    <strong>
      name
    </strong>
     | 
    <strong>
      $
      budget
    </strong>
  </p>
  <Connect(Expenses) />
</div>
`;
